### PR TITLE
action_card: make timeline offset base on start time

### PR DESCRIPTION
**Pull request by sluongng**
Make the action timeline look more like a trace view.

### CODE REVIEW

**Comment by sluongng** [comment]
image

CSS is hard

**Comment by sluongng** [comment]
image

Ok CSS is still hard.

**Comment by bduffany** [comment]
Having to hover to get the timing info feels like a step backward in terms of usability. I also feel like the bars look a bit awkward because they are sort of floating in space, whereas the current gray-ish background provides a nice visual guide.

wdyt about keeping the current chart design but just sliding over the filled regions to correspond with their start time, so that it looks more like a timeline, instead of having them left-aligned like they are today?

**Comment by sluongng** [comment]
I think you are right. There is a downgrade of glanceability now that I have the time to step away and come back to look at this. 🤔 

Let me try the old design with the bars shifted like you suggested.

**Comment by sluongng** [comment]
image

Much simpler change as well 😆

**Comment by sluongng** [comment]
@bduffany what do you think about adding the total duration of the action to the timeline though? We don't have that information from the page, so the user would have to add up the phases manually, or back out to the "Executions" tab to see the total duration there.

**Comment by bduffany** [comment]
> what do you think about adding the total duration of the action to the timeline

sgtm

**Review by bduffany — APPROVED** [review]

File: app/invocation/invocation_action_card.tsx [hunk 1]
```diff
@@ -335,41 +335,61 @@ export default class InvocationActionCardComponent extends React.Component<Props
...
+            style={{ flex: `1 0 0`, backgroundColor: "green" }}></div>
+        </div>
         {filteredEvents.map((event, i) => {
           // Don't render the end marker.
           if (i == filteredEvents.length - 1) return null;
 
           const next = filteredEvents[i + 1];
           const duration = durationSeconds(event.timestamp!, next.timestamp!);
           const weight = duration / totalDuration;
+          offset += weight;
```
Comment (bduffany) [inline app/invocation/invocation_action_card.tsx:368]: sorry I realized I forgot to double-check this math :face_with_spiral_eyes: should the `offset += weight` be happening after we render the element? seems to me like the rough pseudo-code should be

```
weight = duration / totalDuration;
bar1Weight = offset;
bar2Weight = weight;
bar3Weight = 1 - weight - offset;
// Offset should apply to *next* bar that is rendered
offset += weight;
// now render bars with bar1Weight, bar2Weight, bar3Weight
```

Comment (sluongng) [reply app/invocation/invocation_action_card.tsx:368]: Well spot, I thought of adding it after as well, but that would create a larger diff.

So right now I simply add `weight` into `offset` before the loop. Then inside the loop, I subtract out the `weight` value from each bar, then simplified the calculation. Which resulted in the same value.

This is what it currently looks like on my end 
image

**Pull request merged**